### ISSUE TITLE
Some API requests are not forwarded to leader when vault is set in HA mode

### ISSUE THREAD

**Issue by jpbaruteu** [hashicorp/vault#10771]
**Describe the bug**
Hello,

I’m running a vault v1.5.6 HA cluster (with raft storage and tls enabled) in k8s composed by 5 nodes.
The cluster is initialized and all nodes are part of the raft cluster (1 leader and 4 followers). Vault UI is well working and almost all api requests too.

But, I encounter some troubles with few api requests:

API request for vault snapshot

```
curl
–request GET
–header “X-Vault-Token: <VAULT_TOKEN>”
<VAULT_URL>/v1/sys/storage/raft/snapshot > /<SNAPSHOT_NAME>
```

API request to get telemetry metrics

```
curl
–request GET
–header “X-Vault-Token: <VAULT_TOKEN>”
<VAULT_URL>/v1/sys/metrics?format=prometheus
```

At this step:

- <VAULT_URL> is the K8s ingress url wich point to vault service
- <VAULT_TOKEN> is the vault root token which is not revoked as it is a vault test instance

Behaviour:
When I request vault API for those two endpoints, the request is not sent to leader node by the followers and I get this error response:

> {“errors”:[“error performing token check: Vault is in standby mode”]}

Sometimes, the request well responds and I think it is when the request is directely processed by the leader node.
I cannot see any error in vault logs.

Do you have an idea why I face this behaviour ?

Thank you for your help

**To Reproduce**
Run api request on snapshot or metrics endpoint on Vault HA cluster

**Expected behavior**
Requests should be forwarded to the cluster leader.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.5.6
* Vault CLI Version (retrieve with `vault version`): using curl command
* Server Operating System/Architecture: k8s, HA raft cluster with internal storage

**Comment by Ayush-Walia** [comment]
I am also facing exact same issue. @jpbaruteu Did you find any workaround yet?

**Comment by jpbaruteu** [comment]
Hello Ayush,

Unfortunatly no workaround  found for the moment. 
I was hoping someone faced the same issue and have a solution. Maybe this is a bug while runing on k8s ? Strange...

**Comment by Ayush-Walia** [comment]
@jpbaruteu, I found one workaround for my use case, actually I wanted to scrape metrics from vault running in HA, both vault and prometheus are running in k8s in same cluster and I have deployed vault using official helm chart. It turnes out helm charts create one service which will always point to active node "vault-active", so I can use that to always get metrics.

https://github.com/hashicorp/vault-helm/blob/master/templates/server-ha-active-service.yaml

**Comment by jpbaruteu** [comment]
@Ayush-Walia, yes you are rigth. This service always points to the cluster leader and can be used internally if your prometheus is on the same cluster.

But, if your prometheus is external to your cluster, I found that it's possible to enable `/v1/sys/metrics` api endpoint on standby nodes using `unauthenticated_metrics_access` parameter :

```yaml
  ha:
    raft:
      config: |
        ui = true

        listener "tcp" {
          telemetry {
            unauthenticated_metrics_access = true
          }
        }
        telemetry {
        	prometheus_retention_time = "5m",
        	disable_hostname = true
        }
```

Another way could be also to configure kubernetes ingress to point to the `vault-active` service. This allow to solve issue with api request on snapshot endpoint.
By this way, only the cluster leader will be requested by external calls. Don't know if it could potentially generate a bottleneck. I don't think so as finally only the master handles request...

**Comment by heatherezell** [comment]
Thanks for the report! This issue was filed against Vault `v1.5.6`, which is no longer a supported release.

We're unable to investigate bugs against unsupported versions. If you're still experiencing this on a current supported release of Vault, please open a new issue with updated environment details and we'll take a look.

For a list of currently supported versions, see the [Vault version support page](https://github.com/hashicorp/vault/blob/main/.release/versions.hcl). For Enterprise long-term support information, see the [LTS tracker](https://developer.hashicorp.com/vault/docs/updates/lts-tracker).
